File: src/backend/fiscal/src/model/simples_nacional_detalhe_model.py
from src import db


class SimplesNacionalDetalheModel(db.Model):
    __tablename__ = 'simples_nacional_detalhe'

    id = db.Column(db.Integer, primary_key=True)
    faixa = db.Column(db.Integer)
    valor_inicial = db.Column(db.Float)
    valor_final = db.Column(db.Float)
    aliquota = db.Column(db.Float)
    irpj = db.Column(db.Float)
    csll = db.Column(db.Float)
    cofins = db.Column(db.Float)
    pis_pasep = db.Column(db.Float)
    cpp = db.Column(db.Float)
    icms = db.Column(db.Float)
    ipi = db.Column(db.Float)
    iss = db.Column(db.Float)
    id_simples_nacional_cabecalho = db.Column(db.Integer, db.ForeignKey('simples_nacional_cabecalho.id'))


    def mapping(self, data):
        self.id = data.get('id')
        self.id_simples_nacional_cabecalho = data.get('idSimplesNacionalCabecalho')
        self.faixa = data.get('faixa')
        self.valor_inicial = data.get('valorInicial')
        self.valor_final = data.get('valorFinal')
        self.aliquota = data.get('aliquota')
        self.irpj = data.get('irpj')
        self.csll = data.get('csll')
        self.cofins = data.get('cofins')
        self.pis_pasep = data.get('pisPasep')
        self.cpp = data.get('cpp')
        self.icms = data.get('icms')
        self.ipi = data.get('ipi')
        self.iss = data.get('iss')

    def serialize(self):
        return {
            'id': self.id,
            'idSimplesNacionalCabecalho': self.id_simples_nacional_cabecalho,
            'faixa': self.faixa,
            'valorInicial': self.valor_inicial,
            'valorFinal': self.valor_final,
            'aliquota': self.aliquota,
            'irpj': self.irpj,
            'csll': self.csll,
            'cofins': self.cofins,
            'pisPasep': self.pis_pasep,
            'cpp': self.cpp,
            'icms': self.icms,
            'ipi': self.ipi,
            'iss': self.iss,
        }

File: frontend/fiscal/lib/app/data/model/simples_nacional_detalhe_model.dart
import 'dart:convert';
import 'package:pluto_grid/pluto_grid.dart';


class SimplesNacionalDetalheModel {
	int? id;
	int? idSimplesNacionalCabecalho;
	int? faixa;
	double? valorInicial;
	double? valorFinal;
	double? aliquota;
	double? irpj;
	double? csll;
	double? cofins;
	double? pisPasep;
	double? cpp;
	double? icms;
	double? ipi;
	double? iss;

	SimplesNacionalDetalheModel({
		this.id,
		this.idSimplesNacionalCabecalho,
		this.faixa,
		this.valorInicial,
		this.valorFinal,
		this.aliquota,
		this.irpj,
		this.csll,
		this.cofins,
		this.pisPasep,
		this.cpp,
		this.icms,
		this.ipi,
		this.iss,
	});

	static List<String> dbColumns = <String>[
		'id',
		'faixa',
		'valor_inicial',
		'valor_final',
		'aliquota',
		'irpj',
		'csll',
		'cofins',
		'pis_pasep',
		'cpp',
		'icms',
		'ipi',
		'iss',
	];
	
	static List<String> aliasColumns = <String>[
		'Id',
		'Faixa',
		'Valor Inicial',
		'Valor Final',
		'Aliquota',
		'Irpj',
		'Csll',
		'Cofins',
		'Pis Pasep',
		'Cpp',
		'Icms',
		'Ipi',
		'Iss',
	];

	SimplesNacionalDetalheModel.fromJson(Map<String, dynamic> jsonData) {
		id = jsonData['id'];
		idSimplesNacionalCabecalho = jsonData['idSimplesNacionalCabecalho'];
		faixa = jsonData['faixa'];
		valorInicial = jsonData['valorInicial']?.toDouble();
		valorFinal = jsonData['valorFinal']?.toDouble();
		aliquota = jsonData['aliquota']?.toDouble();
		irpj = jsonData['irpj']?.toDouble();
		csll = jsonData['csll']?.toDouble();
		cofins = jsonData['cofins']?.toDouble();
		pisPasep = jsonData['pisPasep']?.toDouble();
		cpp = jsonData['cpp']?.toDouble();
		icms = jsonData['icms']?.toDouble();
		ipi = jsonData['ipi']?.toDouble();
		iss = jsonData['iss']?.toDouble();
	}

	Map<String, dynamic> get toJson {
		Map<String, dynamic> jsonData = <String, dynamic>{};

		jsonData['id'] = id != 0 ? id : null;
		jsonData['idSimplesNacionalCabecalho'] = idSimplesNacionalCabecalho != 0 ? idSimplesNacionalCabecalho : null;
		jsonData['faixa'] = faixa;
		jsonData['valorInicial'] = valorInicial;
		jsonData['valorFinal'] = valorFinal;
		jsonData['aliquota'] = aliquota;
		jsonData['irpj'] = irpj;
		jsonData['csll'] = csll;
		jsonData['cofins'] = cofins;
		jsonData['pisPasep'] = pisPasep;
		jsonData['cpp'] = cpp;
		jsonData['icms'] = icms;
		jsonData['ipi'] = ipi;
		jsonData['iss'] = iss;
	
		return jsonData;
	}
	
	String objectEncodeJson() {
		final jsonData = toJson;
		return json.encode(jsonData);
	}

	plutoRowToObject(PlutoRow plutoRow) {
		id = plutoRow.cells['id']?.value;
		idSimplesNacionalCabecalho = plutoRow.cells['idSimplesNacionalCabecalho']?.value;
		faixa = plutoRow.cells['faixa']?.value;
		valorInicial = plutoRow.cells['valorInicial']?.value?.toDouble();
		valorFinal = plutoRow.cells['valorFinal']?.value?.toDouble();
		aliquota = plutoRow.cells['aliquota']?.value?.toDouble();
		irpj = plutoRow.cells['irpj']?.value?.toDouble();
		csll = plutoRow.cells['csll']?.value?.toDouble();
		cofins = plutoRow.cells['cofins']?.value?.toDouble();
		pisPasep = plutoRow.cells['pisPasep']?.value?.toDouble();
		cpp = plutoRow.cells['cpp']?.value?.toDouble();
		icms = plutoRow.cells['icms']?.value?.toDouble();
		ipi = plutoRow.cells['ipi']?.value?.toDouble();
		iss = plutoRow.cells['iss']?.value?.toDouble();
	}	

	SimplesNacionalDetalheModel clone() {
		return SimplesNacionalDetalheModel(
			id: id,
			idSimplesNacionalCabecalho: idSimplesNacionalCabecalho,
			faixa: faixa,
			valorInicial: valorInicial,
			valorFinal: valorFinal,
			aliquota: aliquota,
			irpj: irpj,
			csll: csll,
			cofins: cofins,
			pisPasep: pisPasep,
			cpp: cpp,
			icms: icms,
			ipi: ipi,
			iss: iss,
		);			
	}

	
}